lib/charms/opensearch/v0/helper_cluster.py

```diff
@@ -58,6 +58,11 @@ def recompute_nodes_conf(nodes: List[Node]) -> Dict[str, Node]:
         if updated_node:
             nodes_by_name[updated_node.name] = updated_node
 
+        # this is in case the nodes had had a change in the "base roles", i.e: removed "ingest"
```

Comment:
Very good catch, thank you! The logic there was wrong: this part of the function essentially handles the case where we are moving from: `user provided roles => automatically generated` 
This logic worked because I tested on "single cluster" deployments, so any node should have contained the `cluster_manager` role. But In a multi cluster deployment this is wrong, I implemented a `refill_node_with_default_roles` function that takes into consideration the multi cluster constraint where the current cluster may not have had `cluster_manager`